### PR TITLE
holdings: display holdings records

### PR DESCRIPTION
**Pull request by BadrAly**
* Adds holdings records display to the document detailed view.

Co-Authored-by: Sebastien Deleze <sebastien.deleze@rero.ch>
Co-Authored-by: Aly Badr <aly.badr@rero.ch>
Co-Authored-by: bertrand Zuchuat <bertrand.zuchuat@rero.ch>

To test:
* display a document detailed view and make sure information about holdings and items are displayed correctly

### CODE REVIEW

**Comment by BadrAly** [comment]
>     * Some documents have empty holdings ([doc 277](https://ilsdev.test.rero.ch/aoste/documents/274), [doc 277](https://ilsdev.test.rero.ch/aoste/documents/274), [doc 284](https://ilsdev.test.rero.ch/aoste/documents/284)). It should be impossible.

it depends on the view.

For example:
([doc 277 aoste](https://ilsdev.test.rero.ch/aoste/documents/274) (one item unavailable in Aoste, one item unavailable at highlands but we can not display the item of highlands).
Should we hide the holding of highlands in this case?

([doc 277 highlands](https://ilsdev.test.rero.ch/highlands/documents/274) (one item available in highlands, one item unavailable at Aoste but we can not display the item of Aoste).
Should we hide the holding of Aoste in this case?

([doc 277 global](https://ilsdev.test.rero.ch/global/documents/274) (two items are displayed).

**Comment by pronguen** [comment]
> (doc 277 aoste (one item unavailable in Aoste, one item unavailable at highlands but we can not display the item of highlands).
> Should we hide the holding of highlands in this case?

Of course. We should display in the views only the corresponding holdings!

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 1]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+                          {% else %}
+                            {{number_items}} item
+                          {% endif %}
+                      </div>
+                      {% if holding.available %}
+                          {% set avail_text = _('available') %}
+                      {% else %}
+                          {% set avail_text = _('not available') %}
+                      {% endif %}
+                      <div class="col-2 availability" style="display: none;">
```
Comment (Garfield-fr) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: Do not use "style" on the element, but a class (Ex: .d-none)

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 2]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+        var id = $(this).data('holdingId');
+        $('#' + id).find('.availability').show();
+      })
+
+      $('.collapse').on('shown.bs.collapse', function () {
+        var id = $(this).data('holdingId');
+        $('#' + id).find('.availability').hide();
+      })
+    });
+  </script>
```
Comment (Garfield-fr) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: Put the javascript code in a separate file (e.g. documents/static/js/rero_ils/documents_jq.js) and add it to the bundles.py file

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 3]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+                            {{number_items}} items
+                          {% else %}
+                            {{number_items}} item
+                          {% endif %}
+                      </div>
+                      {% if holding.available %}
+                          {% set avail_text = _('available') %}
+                      {% else %}
+                          {% set avail_text = _('not available') %}
+                      {% endif %}
```
Comment (Garfield-fr) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: Remove line 219 to 223

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 4]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+                      <div class="col-2 availability" style="display: none;">
+                          {% if number_items > 0 %}
+                            <i class="fa fa-circle text-{{ 'success' if holding.available else 'danger' }}"></i> {{ avail_text }}
```
Comment (Garfield-fr) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: Remove variable `avail_text` and add this `{{ _('available') if holding.available else _('not available') }}`

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 5]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+                      </div>
+                      <div class="col-3">
+                          {{ holding|holding_circulation_category}}
+                      </div>
+                      <div class="col-2">
+                          {% if items|length > 1 %}
+                            {{number_items}} items
+                          {% else %}
+                            {{number_items}} item
+                          {% endif %}
```
Comment (Garfield-fr) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: Remove line 213 to 217 and add `{{ number_items }} {{ _('items') if items|length > 1 else _('item') }}`

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html [hunk 6]
```diff
@@ -172,127 +169,159 @@ <h1 class="col-sm-10 col-md-11">{{ record.title }}</h1>
...
+            </ul>
+          {%- endif %}
+      </footer>
+
 {%- endblock body %}
 
 {%- block javascript %}
   {{ super() }}
   {% assets "rero_ils_documents_detailed_js" %}<script src="{{ ASSET_URL }}"></script>{% endassets %}
+  <script>
```
Comment (jma) [inline rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: this should be in a separate js file and included in the bundle

File: rero_ils/modules/items/views.py [hunk 7]
```diff
@@ -127,8 +127,10 @@ def patron_request(viewcode, item_pid=None, pickup_location_pid=None):
 @blueprint.app_template_filter()
 def not_available_reasons(item):
     """Returns reasons why item is not available."""
-    text = ''
+    if item.available:
+        return item.status
     if not item.available:
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/items/views.py]: fixed.

File: rero_ils/modules/items/views.py [hunk 8]
```diff
@@ -127,8 +127,10 @@ def patron_request(viewcode, item_pid=None, pickup_location_pid=None):
 @blueprint.app_template_filter()
 def not_available_reasons(item):
     """Returns reasons why item is not available."""
-    text = ''
+    if item.available:
+        return item.status
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/items/views.py]: Fixed as discussed

File: rero_ils/modules/holdings/views.py [hunk 9]
```diff
@@ -74,3 +74,21 @@ def holding_loan_condition_filter(holding_pid):
...
     return holding.get_holding_loan_conditions()
+
+
+@blueprint.app_template_filter()
+def holding_location(holding):
+    """Returns holding location name as text."""
+    from ..locations.api import Location
+    location_pid = holding.get('location').get('pid')
+    location = Location.get_record_by_pid(location_pid)
+    text = '{lib}: {loc}'.format(
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/holdings/views.py]: fixed

File: rero_ils/modules/holdings/api.py [hunk 10]
```diff
@@ -148,6 +148,25 @@ def get_holdings_pid_by_document_pid_by_org(cls, document_pid, org_pid):
         for holding in results:
             yield holding.pid
 
+    @classmethod
+    def get_holdings_by_document_by_view_code(cls, document_pid, viewcode):
+        """Returns holding pids by document and view code."""
+        all_holdings = []
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/holdings/api.py]: done

File: rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html (hunk 6 above)
Comment (BadrAly) [reply rero_ils/modules/documents/templates/rero_ils/detailed_view_documents.html]: done

File: rero_ils/modules/holdings/views.py [hunk 11]
```diff
@@ -74,3 +74,21 @@ def holding_loan_condition_filter(holding_pid):
...
+        lib=location.get_library().get('code'), loc=location.get('name'))
+    return text
+
+
+@blueprint.app_template_filter()
+def holding_circulation_category(holding):
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/holdings/views.py]: two different things.

File: rero_ils/modules/holdings/api.py [hunk 12]
```diff
@@ -148,6 +148,20 @@ def get_holdings_pid_by_document_pid_by_org(cls, document_pid, org_pid):
...
+        es_query = HoldingsSearch()\
+            .filter('term', document__pid=document_pid)\
+            .source(['pid'])
+        if (viewcode != current_app.
+                config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE')):
+            org_pid = Organisation.get_record_by_viewcode(viewcode)['pid']
+            es_query.filter('term', organisation__pid=org_pid)
+        all_holdings = [result.pid for result in es_query.scan()]
```
Comment (jma) [inline rero_ils/modules/holdings/api.py]: you can return the array directly, variable is useless

Comment (BadrAly) [reply rero_ils/modules/holdings/api.py]: done

**Pull request merged**